x-pack/plugins/triggers_actions_ui/public/application/sections/alerts_table/hooks/use_columns/use_columns.ts

```diff
@@ -225,6 +209,28 @@ export const useColumns = ({
     [browserFields, columns, defaultColumns, setColumnsAndSave]
   );
 
+  const changedColumns = useMemo(() => {
+    const newColumnIds = defaultColumns.map((column) => column.id);
+    const oldColumnIds = defaultColumnsRef.current.map((column) => column.id);
+    const addedColumnIds = difference(newColumnIds, oldColumnIds);
+    const removedColumnIds = difference(oldColumnIds, newColumnIds);
+    return addedColumnIds.concat(removedColumnIds);
```

Comment:
Although, this seems a little hacky, I do not see any other good alternative without making Contexts in Security Solution. It seems to be working well for me. It will good to add cypress tests for this.